test/tpmtest/tpmtest.c

```diff
@@ -458,6 +460,95 @@ void GetTpmVersion()
     }
 }
 
+// Get the TPM 2.0 commands supported by the TPM.
+TSS2_RC GetCommands( TSS2_SYS_CONTEXT *resMgrSysContext, TPML_CCA **supportedCommands )
+{
+    UINT32 numCommands;
+    TPMI_YES_NO moreData;
+    TPMS_CAPABILITY_DATA capabilityData;
+    TPMA_CC *commandPtr;
+    TSS2_RC rval = TSS2_RC_SUCCESS;
+    UINT32 i;
+
+    // First get the number of commands
+    rval = Tss2_Sys_GetCapability( resMgrSysContext, 0,
+            TPM_CAP_TPM_PROPERTIES, TPM_PT_TOTAL_COMMANDS,
+            1, 0, &capabilityData, 0 );
+    if( rval == TPM_RC_SUCCESS &&
+            capabilityData.capability == TPM_CAP_TPM_PROPERTIES &&
+            capabilityData.data.tpmProperties.count == 1 &&
+            capabilityData.data.tpmProperties.tpmProperty[0].property == TPM_PT_TOTAL_COMMANDS )
+    {
+        numCommands = capabilityData.data.tpmProperties.tpmProperty[0].value;
+    }
+    else
+    {
+        goto returnFromGetCommands;
+    }
+
+    // Allocate memory for them
+    *supportedCommands = (TPML_CCA *)malloc( numCommands * sizeof( TPMA_CC ) + sizeof( UINT32 ) );
```

Comment:
What's the reason for the dynamic allocation here? How badly does a static global TPMS_CAPABILITY_DATA structure bloat the binary?